tests/src/Integration/Action/SystemMailToUsersOfRoleTest.php

```diff
@@ -0,0 +1,239 @@
+use Drupal\user\UserStorageInterface;
+use Psr\Log\LoggerInterface;
+
+
+/**
+ * @coversDefaultClass \Drupal\rules\Plugin\RulesAction\SystemMailToUsersOfRole
+ * @group rules_actions
+ */
+class SystemMailToUsersOfRoleTest extends RulesEntityIntegrationTestBase {
+
+  /**
+   * A mocked logger.
+   *
+   * @var \Psr\Log\LoggerInterface|\Prophecy\Prophecy\ProphecyInterface
+   */
+  protected $logger;
+
+  /**
+   * A mocked mail manager.
+   *
+   * @var \Drupal\Core\Mail\MailManagerInterface|\Prophecy\Prophecy\ProphecyInterface
+   */
+  protected $mailManager;
+
+  /**
+   * A mocked user storage.
+   *
+   * @var \Drupal\user\UserStorageInterface|\Prophecy\Prophecy\ProphecyInterface
+   */
+  protected $userStorage;
+
+  /**
+   * Roles to send emails.
+   *
+   * @var \Drupal\user\Entity\Role[]
+   */
+  protected $roles;
+
+  /**
+   * User to send emails.
+   *
+   * @var \Drupal\user\Entity\User[]
+   */
+  protected $users;
+
+  /**
+   * The action to be tested.
+   *
+   * @var \Drupal\rules\Plugin\RulesAction\SystemMailToUsersOfRole
+   */
+  protected $action;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+    $this->enableModule('user');
+    $test_email='admin@example.com';
+
+    // Create two user roles and one user for each.
+    $uid = 1;
+    foreach (['administrator', 'editor'] as $role_name) {
+
+      // Mock the role entity.
+      $role_mock = $this->prophesize(RoleInterface::class);
+      $role_mock->id()
+        ->willReturn($role_name);
+      $this->roles[] = $role_mock->reveal();
+
+      // Mock the user entity.
+      $user_mock = $this->prophesizeEntity(UserInterface::class);
+      $user_mock->id()->willReturn($uid);
+
+      $user_entity_type = $this->prophesize(EntityTypeInterface::class);
+      $user_mock->getEntityType()
+        ->willReturn($user_entity_type->reveal());
+
+      $user_mock->getPreferredLangcode()
+        ->willReturn(LanguageInterface::LANGCODE_SITE_DEFAULT);
+
+      $user_mock->getEmail()
+        ->willReturn($test_email);
+
+      $this->users[$uid] = $user_mock->reveal();
+      $uid++;
+    }
+
+    $this->logger = $this->prophesize(LoggerInterface::class);
+    $logger_factory = $this->prophesize(LoggerChannelFactoryInterface::class);
+    $logger_factory->get('rules')->willReturn($this->logger->reveal());
+
+    $this->mailManager = $this->prophesize(MailManagerInterface::class);
+    $this->userStorage = $this->prophesize(UserStorageInterface::class);
+
+    // Mocked entityManager.
+    $this->entityManager = $this->prophesize(EntityManagerInterface::class);
+
+    $this->entityManager->getStorage('user')
+      ->willReturn($this->userStorage->reveal());
+
+    $this->entityManager->createInstance()
+      ->willReturn($this->userStorage->reveal());
+
+    // Prepare a user entity type instance.
+    $entity_type = reset($this->users)->getEntityType();
+    $this->entityManager->getDefinitions()
+      ->willReturn($entity_type);
```

Comment:
Problem: now the entity manager only returns the definition for the user entity type, but not for the user role entity type. I think this is not necessary because the parent class RulesEntityIntegrationTestBase already mocks that.